src/app/search-page/search.component.ts

```diff
@@ -0,0 +1,226 @@
+    })
+    const uniqueCategories = findUniqueItems(unfilteredCategories);
+
+    this.categoryList = uniqueCategories;
+  }
+
+  getStatusList() {
+    const unfilteredStatusList: string[] = [];
+
+    this.allProductsBySearch.map(item => {
+      unfilteredStatusList.push(item.status)
+    })
+    const uniqueStatusList = findUniqueItems(unfilteredStatusList);
+
+    this.statusList = uniqueStatusList;
+  }
+
+  ngOnInit(): void {
+    this.query = this.route.snapshot.paramMap.get("query");
+
+    this.filterService.categoryData$
+      .pipe(takeUntil(this.destroy$))
+      .subscribe(category => {
+        this.resetFilterValues();
+        this.selectedCategory = category;
+        this.filteredProducts = findItemsByCategory(this.selectedCategory, this.allProductsBySearch);
+        this.visibleProducts = this.filteredProducts.slice(0, 12);
+      })
+
+    this.filterService.statusData$
+      .pipe(takeUntil(this.destroy$))
+      .subscribe(status => {
+        this.resetFilterValues();
+        this.selectedStatus = status;
+        this.filteredProducts = findItemsByStatus(this.selectedStatus, this.allProductsBySearch);
+        this.visibleProducts = this.filteredProducts.slice(0, 12);
+      })
+
+    this.filterService.releaseDateData$
+      .pipe(takeUntil(this.destroy$))
+      .subscribe(value => {
+        this.selectedReleaseDate = value;
+            if (this.filteredProducts.length === 0) {
+              sortProductsByReleaseDate(this.allProductsBySearch, value);
+              this.visibleProducts = this.allProductsBySearch.slice(0, 12);
+            } else {
+              sortProductsByReleaseDate(this.filteredProducts, value);
+              this.visibleProducts = this.filteredProducts.slice(0, 12);
+            }
+      })
+
+    if (this.query) {
+      this.productService.getProductsByQuery(this.query)
+        .pipe(takeUntil(this.destroy$))
+        .subscribe({
+          next: data => {
+            this.resetFilterValues();
+            this.allProductsBySearch = data.results;
+            this.searchCount = data.count;
+            this.visibleProducts = this.allProductsBySearch.slice(0, 12);
+            this.getCategoryList();
+            this.getStatusList();
+          }
+        })
+    }
+
+     this.queryService.queryData$
+      .pipe(takeUntil(this.destroy$))
+      .subscribe(query => {
+        if (query) {
+          this.productService.getProductsByQuery(query)
+            .pipe(takeUntil(this.destroy$))
+            .subscribe({
+              next: data => {
+                this.resetFilterValues();
+                this.allProductsBySearch = data.results;
+                this.searchCount = data.count;
+                this.query = query;
+                this.visibleProducts = this.allProductsBySearch.slice(0, 12);
+                this.getCategoryList();
+                this.getStatusList();
+              }
+            })
+        }
+      });
```

Comment:
These functions are almost identical.
When this is the case it makes sense to move the identical parts into a new function.

Reply:
Also this part might be similar to the onSearch function on top of this component.

Reply:
- [x] combined into one function